govcd/openapi.go

```diff
@@ -334,6 +334,55 @@ func (client *Client) OpenApiPostItemAndGetHeaders(apiVersion string, urlRef *ur
 	return resp.Header, nil
 }
 
+// OpenApiPostUrlEncoded is a non-standard function used to send a POST request with `x-www-form-urlencoded` format.
+// Accepts a map in format of key:value, marshals the response body in JSON format to outType.
+func (client *Client) OpenApiPostUrlEncoded(urlRef *url.URL, params url.Values, payloadMap map[string]string, outType interface{}, additionalHeaders map[string]string) error {
```

Comment:
I am still thinking if we could actually use existing `OpenApiPostItem` as it looks very similar.

IIRC, the main difference here is the `Content-Type` which by default we assume is `types.JSONMime (application/json)`.

We already have  the patch https://github.com/vmware/go-vcloud-director/pull/577/files#diff-585772f0a826713495acbb0951db5ff9d9f252426fca85e2e112581e5d90a29dR758-R761 that provides mechanism to override `additionalHeaders["Content-Type"] = "application/x-www-form-urlencoded"`

If the only problem remaining now is that `openApiPerformPostPut` tries to "beautify" JSON indentation - we could check if we are overriding `Content-Type` to `application/x-www-form-urlencoded` via `additionalHeader` parameter and skip it if so

Reply:
Well, I think the main point is that without the new function, we would expect the caller function to provide properly encoded string for `x-www-form-urlencoded` instead of a map, like it is now which is IMO more intuitive. I guess we could avoid the new function by creating an if statement, but we should agree on the type we should expect from the user.

Reply:
Ok, agreed. If you could just put a note in that if one provides `Content-Type` in the `additionalHeaders` parameter for `OpenApiPostUrlEncoded` - it will be overriden to `application/x-www-form-urlencoded"` as the behavior is sort of "masked"

Reply:
Added one more line to the function comment, it's also described in the function.

Reply:
Thanks!

> it's also described in the function

Yep, but as an SDK user (and this is a public function) one does not want to read internal function docs unless there is some real trouble